core/toolbox.js

```diff
@@ -184,22 +185,24 @@ Blockly.Toolbox.prototype.init = function() {
       }));
   workspaceOptions.toolboxPosition = workspace.options.toolboxPosition;
 
-  if (workspace.horizontalLayout) {
-    if (!Blockly.HorizontalFlyout) {
-      throw Error('Missing require for Blockly.HorizontalFlyout');
-    }
-    this.flyout_ = new Blockly.HorizontalFlyout(workspaceOptions);
-  } else {
-    if (!Blockly.VerticalFlyout) {
-      throw Error('Missing require for Blockly.VerticalFlyout');
-    }
-    this.flyout_ = new Blockly.VerticalFlyout(workspaceOptions);
+
+  var FlyoutClass = Blockly.registry.getClassFromOptions(
+      Blockly.registry.Type.FLYOUTS_TOOLBOX, this.workspace_.options);
+  var defaultClass = Blockly.registry.getClass(
+      Blockly.registry.Type.FLYOUTS_TOOLBOX, Blockly.registry.DEFAULT)
+
+  if (FlyoutClass == defaultClass && workspace.horizontalLayout) {
+    FlyoutClass = Blockly.registry.getClass(Blockly.registry.Type.FLYOUTS_TOOLBOX,
+        Blockly.HorizontalFlyout.registryName);
```

Comment:
I can see why this is necessary (we want to use a different default based on the horizontalLayout flag). Do you think doing this would make the logic clearer:

- Instead of registering the two flyouts (vertical and horizontal) under the flyouts_toolbox type, register each under a different type (eg: flyouts_vertical_toolbox, flyouts_horizontal_toolbox), that way each has its own default and you're able to keep similar logic that bases which flyout to use on whether the workspace.horizontalLayout is present.

Reply:
Yeah this definitely makes the logic much nicer. If we get to the point where we can move the horizontal flyout into a plugin we can probably get rid of the two names and just use flyout-toolbox.

Reply:
That sounds good.